### PR TITLE
Cvode

### PR DESCRIPTION
**Pull request by MauriceHendrix**
## Description

Added CVODE but without jacob

## Motivation and Context

need to be able to generate cvode models

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have updated all documentation necessary.
- [ ] I have checked spelling in (new) comments.

## Testing
- [X] Testing is done automatically and codecov shows test coverage
- [ ] This cannot be tested automatically 
Please note: just doing 1 model in the tests at the moment only visually verified for now, more to come later.

### CODE REVIEW

File: chaste_codegen/data/tests/console_sctipt_help.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-usage: chaste_codegen [-h] [--version] [-t TYPE] [-o OUTFILE] [-c CLASS_NAME]
-                      [-y] [--expose-annotated-variables]
+usage: chaste_codegen [-h] [--version] [-t TYPE] [-o OUTFILE] [--j]
```
Comment (jonc125) [inline chaste_codegen/data/tests/console_sctipt_help.txt]: File name should have 'script' not 'sctipt' ;)

Comment (jonc125) [reply chaste_codegen/data/tests/console_sctipt_help.txt]: Also, I'd expect just `-j` not `--j`

File: chaste_codegen/cvode_chaste_model.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            'default_stimulus_equations': self._formatted_default_stimulus,
+            'use_get_intracellular_calcium_concentration':
+                self._cytosolic_calcium_concentration_var in self._state_vars,
+            'free_variable': self._free_variable,
+            'use_verify_state_variables': self._use_verify_state_variables,
+            'derived_quantities': self._formatted_derived_quant,
+            'jacobian_equations': []})
+
+        # Generate cpp for model
+        template = cg.load_template('cvode_model.cpp')
```
Comment (jonc125) [inline chaste_codegen/cvode_chaste_model.py]: Did you miss including the template files in this PR?

File: chaste_codegen/_command_line_script.py [hunk 3]
```diff
@@ -28,29 +25,37 @@ def chaste_codegen():
...
+                        'Choices: ' + str(list(translators.keys())))
     parser.add_argument('-o', dest='outfile', metavar='OUTFILE', default=None,
                         help='write program code to OUTFILE '
                              '[default action is to use the input filename with a different extension] '
                              'NOTE: expects provided OUTFILE to have an extension relevant to code being generated '
                              '(e.g. for CHASTE/C++ code: .cpp, .c, .hpp, or .h)')
 
+    group = parser.add_argument_group('Transformations', 'These options control which transformations '
+                                      '(typically optimisations) are applied in the generated code')
+    group.add_argument('--j', '--use-analytic-jacobian', dest='use_analytic_jacobian', default=False,
```
Comment (jonc125) [inline chaste_codegen/_command_line_script.py]: Here, `-j` not `--j`

File: chaste_codegen/data/tests/chaste_reference_models/Cvode/hund_rudy_2004_a.cpp [hunk 4]
```diff
@@ -0,0 +1,899 @@
...
+        const double i_ionic = var_chaste_interface__i_ionic;
+        if (made_new_cvode_vector)
+        {
+            DeleteVector(rY);
+        }
+        EXCEPT_IF_NOT(!std::isnan(i_ionic));
+        return i_ionic;
+    }
+    
+    void Cellhund_rudy_2004_aFromCellML::EvaluateYDerivatives(double var_chaste_interface__Environment__time, const std::vector<double>& rY, std::vector<double>& rDY)
```
Comment (jonc125) [inline chaste_codegen/data/tests/chaste_reference_models/Cvode/hund_rudy_2004_a.cpp]: The .hpp correctly has the last two args as `N_Vector rY, N_Vector rDY`

Comment (jonc125) [reply chaste_codegen/data/tests/chaste_reference_models/Cvode/hund_rudy_2004_a.cpp]: Ah, and this method appears to be defined correctly a second time below!

Comment (MauriceHendrix) [reply chaste_codegen/data/tests/chaste_reference_models/Cvode/hund_rudy_2004_a.cpp]: uups over eager copy-paste :)

**Comment by MichaelClerx** [comment]
I can't review these @MauriceHendrix !

**Pull request merged**